test/state/player_state_test.cpp

```diff
@@ -161,3 +162,133 @@ TEST_F(PlayerStateTest, GetOffsetFromPosition) {
     EXPECT_EQ(offset1, Vec2D(2, 2));
     EXPECT_EQ(offset2, Vec2D(2, 2));
 }
+
+TEST_F(PlayerStateTest, TowerTest) {
+    Tower &tower1 = player_states[0].towers[2];
+    tower1.id = 10;
+    Tower &tower2 = player_states[1].towers[3];
+    tower2.id = 19;
+    Tower tower3(tower1);
+    Tower &tower4 = player_states[0].towers[5];
+    tower4.id = 29;
+
+    EXPECT_NE(tower1, tower2);
+    EXPECT_EQ(tower1, tower3);
+
+    // Initial values
+    EXPECT_EQ(tower4.position, DoubleVec2D(0, 0));
+    EXPECT_EQ(tower4.state, player_state::TowerState::IDLE);
+    EXPECT_EQ(tower4.age, 0);
+    EXPECT_EQ(tower4.hp, 100);
+    EXPECT_EQ(tower4.impact_radius, 0);
+    EXPECT_EQ(tower4.blasting, false);
```

Comment:
Is tower4 necessary? Can't we do all these checks on tower1?

Reply:
Yeah we could. I'll remove tower4 and do the tests on just tower1